tests/nativeapp/test_manager.py

```diff
@@ -247,26 +249,21 @@ def test_get_app_pkg_distribution_in_snowflake(mock_execute, temp_dir, mock_curs
     assert mock_execute.mock_calls == expected
 
 
+@mock.patch(NATIVEAPP_MANAGER_USE_ROLE)
 @mock.patch(NATIVEAPP_MANAGER_EXECUTE)
 def test_get_app_pkg_distribution_in_snowflake_throws_programming_error(
-    mock_execute, temp_dir, mock_cursor
+    mock_execute, mock_use_role, temp_dir, mock_cursor
 ):
 
     side_effects, expected = mock_execute_helper(
         [
-            (
-                mock_cursor([{"CURRENT_ROLE()": "old_role"}], []),
-                mock.call("select current_role()", cursor_class=DictCursor),
-            ),
-            (None, mock.call("use role package_role")),
```

Comment:
hmm, but not executing "use role" at the right time means that the behaviour could be incorrect. Removing boilerplate is good, but aren't we making the test weaker as a result?

Reply:
the idea is we only need to test the behavior of `use_role` once for each occurrence of it. So if we are writing tests for one function (that uses `use_role`) we can write one (or as many needed) test to make sure that `use_role` works as expected. The rest of the tests we write for this function can mock `use_role`.

Reply:
In this example, `use_role` behavior is tested in `test_get_app_pkg_distribution_in_snowflake`. The rest of the tests that are written for `get_app_pkg_distribution_in_snowflake` in particular, can mock `use_role`.

Reply:
For a happy path vs exception path pair of tests, that might make sense in some cases, but I'm not sure I'm generally aligned with this idea. And even then, it seems like a micro-optimization to me. As Cam was saying earlier today, DRY isn't necessary a target in tests, and often will end up with negative side effects. Happy to discuss more when I'm back if you want.